Users/DarkfyreAlgoma/material-api/FixedMaterialDemo.cs

```diff
@@ -0,0 +1,230 @@
+    [DefaultValue("smoothstep")]
+    public readonly string Interpolation;
+
+    #endregion
+    public interface ISimpleData
+    {
+        AgentInfo AgentInfo { get; }
+        ObjectId ObjectId { get; }
+        ObjectId SourceObjectId { get; }
+
+        // Extra data
+        Reflective ExtraData { get; }
+    }
+
+    private MeshComponent Mesh = null;
+
+    public override void Init()
+    {
+
+
+        if (!ObjectPrivate.TryGetFirstComponent(out Mesh))
+        {
+            Log.Write(LogLevel.Error, LogTag, "No MeshComponent found!  Aborting.");
+            return;
+        } else
+        {
+            List<RenderMaterial> materials = Mesh.GetRenderMaterials().ToList();
+            if (materials.Count == 0)
+            {
+                Log.Write(LogLevel.Error, LogTag, "GetRenderMaterials() == null! Aborting.");
+                return;
+            }
+            else
+            {
+                foreach (RenderMaterial material in materials)
+                {
+                    if (material == null)
+                    {
+                        Log.Write(LogTag, "Material is null");
+                    }
+                    else
+                    {
+                        Log.Write(LogTag, material.Name);
+                        Log.Write(LogTag, material.ToString());
+                    }
+                }
+            }
+
+            Log.Write(LogTag, $"Mesh.IsScriptable={Mesh.IsScriptable}");
+
+            if (!Mesh.IsScriptable)
+            {
+                Log.Write(LogLevel.Warning, LogTag, $"MeshComponent {Mesh.Name} is not scriptable");
+            }
+            
+            SubscribeToScriptEvent(AbsorptionCommand, (ScriptEventData data) =>
+            {
+                ISimpleData idata = data.Data.AsInterface<ISimpleData>();
+
+                RenderMaterial m = Mesh.GetRenderMaterial(materials[0].Name);
+                MaterialProperties p = m.GetProperties();
+                
+                p.Absorption = Absorption;
+                m.SetProperties(p, Duration, InterpolationModeParse(Interpolation));
+            });
+            
+            SubscribeToScriptEvent(BrightnessCommand, (ScriptEventData data) =>
+            {
+                ISimpleData idata = data.Data.AsInterface<ISimpleData>();
+
+                RenderMaterial m = Mesh.GetRenderMaterial(materials[0].Name);
+                MaterialProperties p = m.GetProperties();
+                
+                p.Brightness = Brightness;
+                m.SetProperties(p, Duration, InterpolationModeParse(Interpolation));
+            });
+
+            SubscribeToScriptEvent(TintCommand, (ScriptEventData data) =>
+            {
+                ISimpleData idata = data.Data.AsInterface<ISimpleData>();
+
+                RenderMaterial m = Mesh.GetRenderMaterial(materials[0].Name);
+                MaterialProperties p = m.GetProperties();
+
+                if (!Sansar.Color.TryParse(tint, out p.Tint))
```

Comment:
Attempt to import to Sansar client failed with error CS1503: Argument 1: cannot convert from 'Sansar.Color' to 'string', this has not been fixed!
![GitHubFixedMaterialDemo issue 001](https://user-images.githubusercontent.com/5374257/79265427-5edda680-7e96-11ea-9906-47d9c6ba3b37.png)